ppcs/ppcTODO-signature-named-parameters.md

```diff
@@ -0,0 +1,185 @@
+# Named Parameters in Signatures
+
+## Preamble
+
+    Author:  Paul Evans <PEVANS>
+    Sponsor:
+    ID:      TODO
+    Status:  Exploratory
+
+## Abstract
+
+Adds the ability for subroutine signatures to process named arguments in the form of name-value pairs passed by the caller, in a fashion familiar to existing uses of assignment into a hash.
+
+## Motivation
+
+Perl 5.20 added "subroutine signatures", native syntax for more succinctly expressing the common patterns of processing arguments inbound into a subroutine by unpacking the `@_` array. Rather than writing common styles of code directly, a more compact notation, in a style immediately familiar to users of many other languages, allows Perl to create the required behaviour from that specification.
+
+```perl
+sub f ($x, $y, $z = 123) { ... }
+
+# instead of
+sub f {
+    die "Too many arguments" if @_ > 3;
+    die "Too few arguments" if @_ < 2;
+    my ($x, $y, $z) = @_;
+    $z = 123 if @_ < 3;
+    ...
+}
+```
+
+One common form of argument processing involves passing an even-sized list of key/value pairs and assigning that into a hash within the subroutine, so that specifically named parameters can be extracted from it. There is currently no support from subroutine signature syntax to assist authors in providing such behaviours.
```

Comment:
An example of where this could be used is command-line argument processing. Raku makes the [`MAIN` subroutine](https://docs.raku.org/language/create-cli) special and uses the signature to give a default `getopt`-style interface.